draft-ietf-rats-corim.md

```diff
@@ -1640,12 +1650,24 @@ If the `selection` criteria is not satisfied, then evaluation procedes to the ne
 {::include cddl/intrep-ev.cddl}
 ~~~
 
-| Type | `en` | `el` | `a` | `cm` | `p` |
+| ECT type  | ECT Field       | Requirement |
 |---
-| condition | F | T | F | F | F |
-| selection | F | T | F | F | F |
-| addition  | T | T | T | T | F |
-{: #tbl-ev-ect-optionality title="Mandatory fields for Endorsed Values tuples"}
+| condition | `environment`   | Mandatory   |
+|           | `elements`      | Mandatory   |
+|           | `authority`     | Optional    |
+|           | `cm type`       | n/a         |
+|           | `profile`       | n/a         |
+| selection | `environment`   | Optional    |
+|           | `elements`      | Mandatory   |
+|           | `authority`     | n/a         |
```

Comment:
Why can't authority be optional here to allow for further narrowing of allowed authorities based on the elements?

We're not talking about matching only a single ECT at a time, since we have [ + ECT ] in both selection and condition, so we're always just narrowing the pool, and if the pool is becomes empty, we short-circuit and try the next in the series.

Reply:
Good point. Are there any disadvantages to render authority an optional item here?

Reply:
I marked it n/a based on the expectation that the selection is a subset of the results of the condition match. If authority is a constraint in the condition, applying an authority constraint again would be idempotent (I think). Is there a use case where applying an authority constraint in the selection would produce different results from it having been applied in the condition?

Alternatively, the internal representation could take the path that the selection context is not constrained by the condition. In which case, the selection environment, elements, and authority are mandatory. When mapping the series triple where the selection depends on the condition, the dependent parts will be copied into the selection to produce a seemingly redundant representation.

Reply:
I don't believe it will be idempotent if you have a common condition that mkey 0 has measurements signed by X, but you have mkey 1 signed by Y, Z, or W leading to different outcomes.

Reply:
commit https://github.com/ietf-rats-wg/draft-ietf-rats-corim/pull/296/commits/4ab1661542186dde9ffe322d5b4a21e7089221fd changed selection environment to mandatory and selection authority to optional.

The current transformation text for series triple already populates environment but ignores authority since a use case for it is not known to exist.